### ISSUE TITLE
Strings handed to Call parameters get double escaped

### ISSUE THREAD

**Issue by Belgabor** [ironthree/dxr#5]
I need to pass an url with parameters via XML-RPC, unfortunately '&' gets double escaped.

```rust
let call: Call<_, i32> = Call::new("test", vec![Value::string("a&b")]);
client.call(call).await?;
```

From the server log:
```xml
<?xml version="1.0"?>
<methodCall>
  <methodName>test</methodName>
  <params>
    <param>
      <value>
        <string>a&amp;amp;b</string>
      </value>
    </param>
  </params>
</methodCall>
```

**Comment by decathorpe** [comment]
Huh, that does look wrong :(
I'm pretty sure quick-xml didn't do automatic escaping of strings when I wrote the initial implementation, but it looks like it definitely does do now ... I'll look into it. I might be able to drop the manual escaping code if quick-xml now handles this internally

**Comment by decathorpe** [comment]
I've dropped the code that caused the duplicate (un)escaping and  added two small test cases to ensure this won't happen again by accident. I released a new version of the v0.5 branch ([v0.5.4](https://github.com/ironthree/dxr/releases/tag/0.5.4)), which should be available from crates.io any second now. I'll now forward port this fix to the main branch as well. Thanks for the report!
